Scripts/Mac/Tracing/Tracing.md

```diff
@@ -0,0 +1,78 @@
+# Tracing on macOS
+
+[DTrace](https://en.wikipedia.org/wiki/DTrace) is the main low-level tool for tracing and diagnostics on macOS.
+For diagnosing issues potentially caused by VFS for Git, tracing syscalls to track down errors has proven to be a useful technique on Windows.
+On macOS, the DTrace script `dtruss` which ships with the OS can be used for tracing system calls made by processes.
+However, it has not been updated for newer OS versions for a while, making some of its features unreliable.
+We therefore recommend using our [updated version of the script](./dtruss), and the instructions below assume you are using this version.
+
+## TL;DR: This build command is failing/misbehaving and I've been asked to collect a trace
+
+To capture a trace of the syscalls made by the failing command, run the following:
+
+    ++path-to-vfs4g-source++/Scripts/Mac/Tracing/dtruss -d -e -F -f    ++command-to-run++   2> >(tee   ++trace-filename.txt++   >&2)
```

Comment:
I used "2>&1 | tee log" as an alternative

Reply:
Note that while simpler, this also includes stdout in the log file, whereas the code I quoted only saves stderr to the file. This may help keep the output from the command itself separate from the trace data.